flytekit/annotated/base_task.py

```diff
@@ -33,6 +35,60 @@ def kwtypes(**kwargs) -> Dict[str, Type]:
     return d
 
 
+@dataclass
+class TaskMetadata(object):
+    """
+    Create Metadata to be associated with this Task
+
+    Args:
+      cache: Boolean that indicates if caching should be enabled
+      cache_version: Version string to be used for the cached value
+      interruptable: Boolean that indicates that this task can be interrupted and/or scheduled on nodes
+                           with lower QoS guarantees. This will directly reduce the `$`/`execution cost` associated,
+                            at the cost of performance penalties due to potential interruptions
+      deprecated: A string that can be used to provide a warning message for deprecated task. Absence / empty str
+                       indicates that the task is active and not deprecated
+      retries: for retries=n; n > 0, on failures of this task, the task will be retried at-least n number of times.
+      timeout: the max amount of time for which one execution of this task should be executed for. If the execution
+                     will be terminated if the runtime exceeds the given timeout (approximately)
+     """
+
+    cache: bool = False
+    cache_version: str = ""
+    interruptable: bool = False
+    deprecated: str = ""
+    retries: int = 0
+    timeout: Optional[Union[datetime.timedelta, int]] = None
+
+    def __post_init__(self):
+        if self.timeout:
+            if isinstance(self.timeout, int):
+                self.timeout = datetime.timedelta(seconds=self.timeout)
+            elif not isinstance(self.timeout, datetime.timedelta):
+                raise ValueError("timeout should be duration represented as either a datetime.timedelta or int seconds")
+        if self.cache and not self.cache_version:
```

Comment:
what if cache version is set but cache isn't?

Reply:
I think this happens a lot of times today (at lyft)